api/controllers/Irc_EntryController.js

```diff
@@ -117,31 +119,97 @@ module.exports = {
     switch (request_body.operation) {
     case Event.OPERATION_INTER_SITE_TRANSFER:
       return processEventDetainee(request_body)
-          .then((detainee) => this.handleInterSiteTransfer(detainee, request_body));
+        .then((detainee) => this.handleInterSiteTransfer(detainee, request_body));
     case Event.OPERATION_UPDATE:
       return processEventDetainee(request_body)
-          .tap((detainee) => Centres.publishUpdateOne(detainee.centre));
+        .tap((detainee) => Centres.publishUpdateOne(detainee.centre));
     case Event.OPERATION_CHECK_IN:
     case Event.OPERATION_CHECK_OUT:
     case Event.OPERATION_REINSTATEMENT:
       return processEventDetainee(request_body)
-          .then((detainee) => Event.create(generateStandardEvent(detainee, request_body)))
-          .tap((event) => Centres.publishUpdateOne(event.centre));
+        .then((detainee) => Event.create(generateStandardEvent(detainee, request_body)))
+        .tap((event) => Centres.publishUpdateOne(event.centre));
+    case BedEvent.OPERATION_OUT_OF_COMMISSION:
+    case BedEvent.OPERATION_IN_COMMISSION:
+      return this.processBedEvent(request_body)
+        .tap((event) => Centres.publishUpdateOne(event.centre_id));
     default:
       throw new ValidationError('Unknown operation');
     }
   },
 
+  findAndPopulateCentre: event =>
+    Centres.getByName(event.centre)
+      .then((result) => _.merge(event, {centre_id: result.id})),
+
+  findAndPopulateDetainee: (event) => {
+    let detainee;
+    if (_.isNull(event.single_occupancy_person_id) || _.isUndefined(event.single_occupancy_person_id)) {
+      detainee = Promise.resolve(_.merge(event, {detainee: null}));
+    } else {
+      detainee = createOrUpdateDetainee({
+        person_id: event.single_occupancy_person_id,
+        centre: {id: event.centre_id},
+        timestamp: new Date().toString()
+      })
+        .then((result) =>
+          _.merge(event, {detainee: result.id}));
+    }
+    return detainee;
+  },
+
+  formatAndPopulateGender: (event) =>
+    _.merge(event, {gender: Bed.normalizeGender(event.gender)}),
+
+  findOrCreateAndPopulateBed: (event) =>
+    Bed.findOrCreate(
+      {bed_ref: event.bed_ref, centre: event.centre_id},
+      {
+        centre: event.centre_id,
+        bed_ref: event.bed_ref,
+        gender: event.gender
+      })
+      .then((bed) => _.merge(event, {bed_id: bed.id})),
+
+  findAndPopulateActiveStatus: (event) =>
+    BedEvent.findOne({bed: event.bed_id, timestamp: {'>=': event.timestamp}})
+      .then((result) =>
+        _.merge(event, {active: _.isUndefined(result)})),
+
+  reconcilePreviousBedEvents: (event) =>
+  event.active && BedEvent.deactivatePastBedEvents(event.bed_id, event.timestamp),
```

Comment:
indentation: this has been discussed in previous PR's -> I am able to improve indention by adding parenthesis around the expression; however, eslint setting currently through an 'error' stating 'gratuitous parenthesis around expression.' Previously, we agreed that 'incorrect' indentation was allowed in these circumstances. 

Passing the centre into whatever this method does: This is not required, as the bed id is passed, which has a reference (FK) to the centre.